google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -2048,6 +2048,7 @@ boolean isClosed() {
 
     // Does various pool maintenance activities.
     void maintainPool() {
+      Instant currTime = clock.instant();
```

Comment:
nit: Do we really have to initialize here? Can we initialize before if statement inside synchronized block so that we will get exact current time. It would be okay to initialize here if we don't have synchronized block.

Reply:
I just noticed that go does the same way where they are assigning right before if statement. https://github.com/googleapis/google-cloud-go/blob/main/spanner/session.go#L1775

Reply:
this `currTime` is used at multiple places. Inside synchronized block and outside as well. It will be an overkill if we initialize at multiple times. Inside block and outside. WDYT ?